src/App.tsx

```diff
@@ -1,24 +1,71 @@
 import React from 'react';
 import './App.scss';
 
-const App: React.FC = () => {
-  setInterval(() => {
+type Props = {
+
+};
+
+type State = {
+  isClockVisible: boolean,
+  clock: string,
+};
+
+class App extends React.Component<Props, State> {
+  state = {
+    isClockVisible: true,
+    clock: '',
+  };
+
+  componentDidMount() {
+    setInterval(() => {
+      if (this.state.isClockVisible) {
```

Comment:
There's no need to have some logic of hiding/showing clock inside setInterval method.
You have hideClock and showClock methods for this